src/examples/tls_stream_coroutine_client.cpp

```diff
@@ -3,10 +3,17 @@
 
 #include <botan/asio_compat.h>
 
+#if !defined(BOTAN_HAS_HAS_DEFAULT_TLS_CONTEXT)
+
+int main() {
+   std::cout << "Your system needs an auto seeded RNG and a certificate store.\n";
+   return 1;
+}
+
 // Boost 1.81.0 introduced support for the finalized C++20 coroutines
 // in clang 14 and newer. Older versions of Boost might work with other
 // compilers, though.
-#if defined(BOTAN_FOUND_COMPATIBLE_BOOST_ASIO_VERSION) && BOOST_VERSION >= 108100
+#elif defined(BOTAN_FOUND_COMPATIBLE_BOOST_ASIO_VERSION) && BOOST_VERSION >= 108100
```

Comment:
Perhaps just check for the default TLS context macro here as well and update the fallback-error message at the very end of the example to be more ambigous.